scripts/breaking_changes_checker/breaking_changes_tracker.py

```diff
@@ -103,13 +103,30 @@ def __init__(self, stable: Dict, current: Dict, package_name: str, **kwargs: Any
         for checker in checkers:
             if not isinstance(checker, ChangesChecker):
                 raise TypeError(f"Checker {checker} does not implement ChangesChecker protocol")
+        post_processing_checkers: List[PostProcessingChecker] = kwargs.get("post_processing_checkers", [])
+        for checker in post_processing_checkers:
+            if not isinstance(checker, PostProcessingChecker):
+                raise TypeError(f"Checker {checker} does not implement PostProcessingChecker protocol")
         self.checkers = checkers
+        self.post_processing_checkers = post_processing_checkers
 
     def run_checks(self) -> None:
         self.run_breaking_change_diff_checks()
         self.check_parameter_ordering()  # not part of diff
-        self.run_async_cleanup(self.breaking_changes)
+        self.run_post_processing()
 
+    def run_post_processing(self) -> None:
+        # Remove duplicate reporting of changes that apply to both sync and async package components
+        self.run_async_cleanup(self.breaking_changes)
```

Comment:
In a separate PR this will be cleaned up to just be another checker that's passed in